zebra-rpc/src/methods/get_block_template_rpcs.rs

```diff
@@ -333,40 +333,113 @@ where
             // These checks always have the same result during long polling.
             let miner_address = check_miner_address(miner_address)?;
 
+            let mut client_long_poll_id = None;
             if let Some(parameters) = parameters {
-                check_block_template_parameters(parameters)?;
+                check_block_template_parameters(&parameters)?;
+
+                client_long_poll_id = parameters.long_poll_id;
```

Comment:
I think we should block this method here by awaiting `wait_for_max_time`, a changed `latest_chain_tip`, or a changed `mempool_transaction_receiver`, and leave the rest of this function as it is.

Reply:
I'm not sure if this would work the way it is meant to?

The long polling spec says that we need to return immediately if the client long poll ID is different from the current server long poll ID:
> This request SHOULD NOT be processed nor answered by the server until it wishes to replace the current block data as identified by the "longpollid".

https://en.bitcoin.it/wiki/BIP_0022#Optional:_Long_Polling

Using watch channels is also out of scope for this PR, see the TODOs below.

Reply:
> I'm not sure if this would work the way it is meant to?

Ah, you're right, it needs to check that the long poll id is the same first.